src/core/PointerLockControls.tsx

```diff
@@ -44,9 +44,11 @@ export const PointerLockControls = React.forwardRef<PointerLockControlsImpl, Poi
 
     React.useEffect(() => {
       const handler = () => controls?.lock()
-      const element = selector ? document.querySelector(selector) : document
-      element && element.addEventListener('click', handler)
-      return () => (element ? element.removeEventListener('click', handler) : undefined)
+      const elements = selector ? Array.from(document.querySelectorAll(selector)) : [document]
+      elements.map((element) => element && element.addEventListener('click', handler))
```

Comment:
why are we mapping? We could just `forEach` it? same with the return function.

Reply:
Hmm, I decided to use mapping as I've always thought it to be faster (which was apparently a lie) and just more elegant solution.

Learned something new today and changed it to forEach :)